pkg/task/manager.go

```diff
@@ -102,6 +108,18 @@ func (m *manager) Finish(id string, err error) {
 	}
 }
 
+func (m *manager) SetMetric(id string, metric *converter.Metric) {
```

Comment:
Can we fill the metric in `Finish`?

Reply:
Due to concurrency constraints, we can't fill metric on all tasks based on `singleflight`, because `singleflight` only executes once but the `taskID` for each task is different.